platforms/defaults_windows.go

```diff
@@ -63,6 +65,10 @@ func (m windowsmatcher) Less(p1, p2 specs.Platform) bool {
 	m1, m2 := m.Match(p1), m.Match(p2)
 	if m1 && m2 {
 		r1, r2 := revision(p1.OSVersion), revision(p2.OSVersion)
+		mubr1, mubr2 := r1 == GetCurrentWindowsUpdateBuildRevision(), r2 == GetCurrentWindowsUpdateBuildRevision()
```

Comment:
This is just another level of comparison, similar to OSVersion. This function was never a strict less, but a function to order preference. Maybe it should be renamed?

Reply:
I think this is the right spot. Less is the interface so we need it keep it. But I agree its for sorting the precedence we need to agree on the sort order. I commented below about that as well.

Reply:
While this is the right place to put this logic, I think relying on the _current system_'s version is a bit weird here, when we're sorting rather than filtering (and _only_ doing it for patch level, not the rest). `Less` should behave consistently with its inputs, i.e. if we want to deal with UBR this way, it should be added to the `windowsmatcher`. not fetched here.

I've also left further comments on the test-case which would change this completely if enacted.